src/OrchardCore.Modules/OrchardCore.Notifications/Drivers/NotificationNavbarDisplayDriver.cs

```diff
@@ -46,7 +51,21 @@ public override IDisplayResult Display(Navbar model, BuildDisplayContext context
             model.TotalUnread = notifications.Count;
 
         }).Location("Detail", "Content:9")
-        .Location("DetailAdmin", "Content:9")
-        .RenderWhen(() => _authorizationService.AuthorizeAsync(_httpContextAccessor.HttpContext.User, NotificationPermissions.ManageNotifications));
+        .Location("DetailAdmin", "Content:9");
+
+        if (_notificationOptions.CacheDurationInSeconds > 0)
+        {
+            var username = _httpContextAccessor.HttpContext.User.Identity.Name;
+
+            return result
+                .Cache(NotificationConstants.TopUnreadUserNotificationCacheTag, context => context
+                    .AddContext("user")
+                    .WithExpiryAfter(TimeSpan.FromSeconds(_notificationOptions.CacheDurationInSeconds.Value))
+                    .AddTag(NotificationConstants.TopUnreadUserNotificationCacheTag)
```

Comment:
This tag is never removed, maybe not use it then?

Reply:
It's never used correct. I kept it just in case someone wants to remove all associated with notification. One may want to create UI based configuration that would require clearing the cache from all users.